script.js

```diff
@@ -0,0 +1,235 @@
+    },
+  ]
+}
+
+/*------ C O N T R O L E R ------*/
+ var controler = {
+    init: function () {
+      modal.currentSlide = modal.slides[0];
+      
+      viewSlide.init();
+      viewThumbnail.init();
+      viewOptionBar.init();
+    },
+
+    getCurrentSlide: function () {
+      return modal.currentSlide;
+    },
+
+    setCurrentSlide: function (slide) {
+      modal.currentSlide = slide;
+    },
+
+    getSlides: function () {
+      return modal.slides;
+    },
+
+    getNextSlide: function () {
+      var nextIdx = modal.currentSlide.idx;
+      nextIdx++;
+
+      if(nextIdx >= modal.slides.length){
+        nextIdx = 0;
+      }
+
+      return modal.slides[nextIdx];
+    },
+
+    getPrevSlide: function () {
+      var prevIdx = modal.currentSlide.idx;
+      prevIdx--;
+      if(prevIdx < 0){
+        prevIdx = (modal.slides.length-1);
+      }
+
+      return modal.slides[prevIdx];
+    },
+
+    getCurrentSlideIndex: function () {
+      return modal.currentSlide.idx;
+    },
+    
+    getAutoPlay: function () {
+      return modal.autoPlay;
+    },
+
+    setAutoPlay: function (bool) {
+      modal.autoPlay = bool;
+    },
+
+    getSLIDE_TIMEOUT: function () {
+      return  modal.SLIDE_TIMEOUT;
+    }
+ };
+
+ /*------ V I E W ------*/
+ var viewSlide = {
+    init: function () {
+
+      this.slideElem = document.querySelector('.slideshow__slide');
+      this.slideTitleElem = document.querySelector('.slideshow__title');
+      this.slideImgElem = document.querySelector('#slideshow__img');
+      this.slideDescriptionElem = document.querySelector('.slideshow__paragraph');
+      this.slideReferenceElem = document.querySelector('.slideshow__reference');
+
+      this.slideTitleElem.addEventListener('click', function () {
+        viewSlide.slideReferenceElem.style.display = 
+              viewSlide.slideReferenceElem.style.display === 'none'? 'block' : 'none';
+      })
+
+      this.render();
+    },
+
+    render: function () {
+      var currentSlide = controler.getCurrentSlide;
+      this.slideTitleElem.firstElementChild.textContent = modal.currentSlide.title;
+      this.slideImgElem.src = modal.currentSlide.imgSrc;
+      this.slideDescriptionElem.firstElementChild.textContent = modal.currentSlide.description;
+      this.slideReferenceElem.textContent = modal.currentSlide.reference;
+      this.slideImgElem.alt = modal.currentSlide.reference;
+    },
+
+    autoRender: function () {
+      var slide = controler.getNextSlide();
+      controler.setCurrentSlide(slide);
+      viewSlide.render();
+      viewOptionBar.render();
+    }
+ };
+
+ var viewThumbnail = {
+  init: function () {
+    this.elemThumbnails = document.querySelector('.slideshow__thumbnails');
+
+    this.render();
+  },
+
+  render: function () {
+    var slides = controler.getSlides();
+
+    for(var i = 0; i < slides.length; i++){
+      slide = slides[i];
+
+      elemDiv = document.createElement('div');
```

Comment:
all of this could be in a function called `setThumbnail`, `addThumbnailListeners` and then `this.elemThumbnails.appendChild(elemDiv)`

Reply:
Where should those go??